index.js

```diff
@@ -279,18 +264,19 @@ class ServerlessCustomDomain {
   getHostedZoneId() {
     const hostedZonePromise = this.route53.listHostedZones({}).promise();
 
-    return hostedZonePromise.then(sortHostedZone)
+    return hostedZonePromise
       .then((data) => {
-        // Gets the hostzone that contains the root of the custom domain name
-        let hostedZoneId = data.HostedZones.find((hostedZone) => {
-          let hZoneName = hostedZone.Name;
-          // Takes out the . at the end if there is one
-          hZoneName = hZoneName.endsWith('.') ? hZoneName.slice(0, -1) : hZoneName;
-          return (this.targetHostedZone || '').endsWith(hZoneName);
-        });
-
-        if (hostedZoneId) {
-          hostedZoneId = hostedZoneId.Id;
+        // Gets the hostzone that is closest match to the custom domain name
+        const targetHostedZone = data.HostedZones
+          .filter((hostedZone) => {
+            const hostedZoneName = hostedZone.Name.endsWith('.') ? hostedZone.Name.slice(0, -1) : hostedZone.Name;
+            return (this.targetHostedZone || '').endsWith(hostedZoneName);
```

Comment:
Its a bit confusing that we have 2 variables with the same name `this.targetHostedZone` and the local variable `targetHostedZone`. Maybe one of those should be renamed. Otherwise this looks good

Reply:
Also when would `this.targetHostedZone` ever be missing?